Library/Homebrew/api/cask.rb

```diff
@@ -20,12 +21,22 @@ def fetch(token)
           Homebrew::API.fetch "cask/#{token}.json"
         end
 
-        sig {
-          params(token: String, path: T.any(String, Pathname), git_head: String,
-                 sha256: T.nilable(String)).returns(String)
-        }
-        def fetch_source(token, path:, git_head:, sha256: nil)
-          Homebrew::API.fetch_homebrew_cask_source token, path: path, git_head: git_head, sha256: sha256
+        sig { params(cask: ::Cask::Cask).returns(::Cask::Cask) }
```

Comment:
are all the `::` necessary? It feels like there's way more used in this PR than I typically see used?

Reply:
Will double check this - it is for `::URL` in `cask/download.rb`.

Reply:
Cool, thanks 🙇🏻

Reply:
This picks up `API::Cask` so is needed.